src/lib/core/tt_static.h

```diff
@@ -56,6 +59,17 @@ tt_static_buf(void)
 					     sizeof(intptr_t));
 }
 
+/**
+ * Return a thread-local statically allocated temporary buffer of
+ * size @a TT_STATIC_BUF_BIG_LEN.
+ */
+static inline char *
+tt_static_buf_big(void)
```

Comment:
I don't quite like the fact that we have two kinds of static buffers now. I think it's possible to format and escape the message right in the `say_buf`. To achieve that, we could:

1. Make `json_escape` work in-place (i.e. with the same `buf` and `data`). AFAIU this should be fairly easy to do: first, scan the input string to count its final size, then escape characters while scanning the input string backwards.
2. Patch `say_format_json` so that it works as follows:
   - Format the string with an empty message.
   - After formatting the log record, `memmove` everything after the message to the end of the buffer.
   - Next, format the message in the resulting gap and escape it in-place.
   - Finally, `memmove` the message tail after the message.

What do you think?